src/components/implementation/cbufp/default/cbufp.c

```diff
@@ -618,14 +616,14 @@ cbufp_retrieve(spdid_t spdid, int cbid, int size)
 	assert(page);
 	if (cbufp_map(spdid, dest, page, size, MAPPING_READ))
 		valloc_free(cos_spd_id(), spdid, (void *)dest, 1);
-
-	meta->nfo.c.flags |= CBUFM_TOUCHED;
-	meta->nfo.c.ptr    = map->addr >> PAGE_ORDER;
-	meta->sz           = cbi->size >> PAGE_ORDER;
-	ret                = 0;
+	memset(meta, 0, sizeof(struct cbuf_meta));
```

Comment:
Note that this will set the contents of meta to `0` starting from the lower bytes, to the higher bytes.  Please confirm that there is no race between this and the client (i.e. the client sees a 0 at one lower-byte field, and does something that gets erased by a concurrent `memset`.